scripts/build/termux_step_setup_variables.sh

```diff
@@ -51,15 +66,43 @@ termux_step_setup_variables() {
 		TERMUX_ARCH_BITS=32
 	fi
 
-	TERMUX_HOST_PLATFORM="${TERMUX_ARCH}-linux-android"
-	if [ "$TERMUX_ARCH" = "arm" ]; then TERMUX_HOST_PLATFORM="${TERMUX_HOST_PLATFORM}eabi"; fi
-
-	if [ "$TERMUX_ON_DEVICE_BUILD" = "false" ] && [ ! -d "$NDK" ]; then
-		termux_error_exit 'NDK not pointing at a directory!'
+	if [ "$TERMUX_PACKAGE_LIBRARY" = "bionic" ]; then
+		TERMUX_HOST_PLATFORM="${TERMUX_ARCH}-linux-android"
+	else
+		TERMUX_HOST_PLATFORM="${TERMUX_ARCH}-linux-gnu"
 	fi
+	if [ "$TERMUX_ARCH" = "arm" ]; then
+		TERMUX_HOST_PLATFORM="${TERMUX_HOST_PLATFORM}eabi"
+		if [ "$TERMUX_PACKAGE_LIBRARY" = "glibc" ]; then
+			TERMUX_HOST_PLATFORM="${TERMUX_HOST_PLATFORM}hf"
+		fi
+	fi
+
+	if [ "$TERMUX_PACKAGE_LIBRARY" = "bionic" ]; then
+		if [ "$TERMUX_ON_DEVICE_BUILD" = "false" ] && [ ! -d "$NDK" ]; then
+			termux_error_exit 'NDK not pointing at a directory!'
+		fi
 
-	if [ "$TERMUX_ON_DEVICE_BUILD" = "false" ] && ! grep -s -q "Pkg.Revision = $TERMUX_NDK_VERSION_NUM" "$NDK/source.properties"; then
-		termux_error_exit "Wrong NDK version - we need $TERMUX_NDK_VERSION"
+		if [ "$TERMUX_ON_DEVICE_BUILD" = "false" ] && ! grep -s -q "Pkg.Revision = $TERMUX_NDK_VERSION_NUM" "$NDK/source.properties"; then
+			termux_error_exit "Wrong NDK version - we need $TERMUX_NDK_VERSION"
+		fi
+	elif [ "$TERMUX_PACKAGE_LIBRARY" = "glibc" ]; then
+		if [ "$TERMUX_ON_DEVICE_BUILD" = "true" ]; then
+			if [ -n "${LD_PRELOAD-}" ]; then
+				unset LD_PRELOAD
+			fi
+			if [ -d "${TERMUX_PREFIX}/bin" ]; then
+				export PATH="${TERMUX_PREFIX}/bin:${PATH}"
+			else
+				termux_error_exit "Glibc components are not installed, run './scripts/setup-termux-glibc.sh'"
+			fi
+		else
+			if [ -d "${CGCT_DIR}/${TERMUX_ARCH}/bin" ]; then
+				export PATH="${CGCT_DIR}/${TERMUX_ARCH}/bin:${PATH}"
+			else
+				termux_error_exit "The cgct tools were not found"
```

Comment:
Would be good to tell the users how to install them. The setup-cgct script is pretty secretive as it looks now

Reply:
ok